### ISSUE TITLE
FPing returns 100% loss in Docker bridge mode on Synology, but works with host networking

### ISSUE THREAD

**Issue by MNTsnowman** [everywan-dev/smokeping#7]
<html>
<body>
<html><head></head><body><p>Hi,</p><p>I'm running <code inline="">sistemasminorisa/smokeping:latest</code> on a Synology NAS DS918+ using Container Manager.</p><p>I have encountered an unusual networking issue where SmokePing works correctly in terms of its web interface and traceroute functionality, but FPing does not receive any ICMP replies when the container uses Docker bridge networking.</p><h2>Environment</h2><ul><li><p>Image: <code inline="">sistemasminorisa/smokeping:latest</code></p></li><li><p>Network mode: Docker bridge</p></li><li><p>Capabilities:</p><ul><li><p><code inline="">NET_RAW</code></p></li><li><p><code inline="">NET_ADMIN</code></p></li></ul></li><li><p>Container port: 80</p></li><li><p>Host port in bridge mode: 8086</p></li><li><p>Volumes:</p><ul><li><p><code inline="">/config</code></p></li><li><p><code inline="">/data</code></p></li></ul></li><li><p>Timezone: <code inline="">Europe/Copenhagen</code></p></li></ul><h2>SmokePing configuration</h2><p>I reduced the targets to only four targets, two on LAN and two on WAN.</p><p>FPing is configured with:</p><pre><code class="language-text">+ FPing
binary = /usr/sbin/fping
step = 60
pings = 20
</code></pre><p>SmokePing starts normally and reports:</p><pre><code class="language-text">FPing: probing 4 targets with step 60 s
Smokeping version 2.009000 successfully launched
</code></pre><p>However, the RRD data contained <code inline="">NaN</code> for the latency values.</p><p>For example:</p><pre><code class="language-text">rrdtool fetch /data/Local/Router.rrd AVERAGE --start end-10min
</code></pre><p>returned entries containing:</p><pre><code class="language-text">-nan 2.0000000000e+01 -nan -nan ...
</code></pre><p>The <code inline="">20</code> corresponds to the configured number of pings.</p><p>I also tested FPing directly from inside the running SmokePing container:</p><pre><code class="language-text">fping -c 4 &lt;router&gt;
</code></pre><p>and consistently received:</p><pre><code class="language-text">xmt/rcv/%loss = 4/0/100%
</code></pre><h2>Interesting observation</h2><p>I then created a copy of the same container using <strong>Host networking</strong> instead of <strong>Bridge</strong>, with the same <code inline="">/config</code> and <code inline="">/data</code> directories.</p><p>The Host-networked container cannot expose its web interface through another Docker port because Apache inside the image listens on port 80 and Host networking bypasses Docker port mapping. Port 80 on the Synology is already occupied by DSM/reverse proxy.</p><p>However, while the Host-networked container was running, the same RRD files began receiving <strong>real latency values instead of NaN</strong>.</p><p>The original container's web interface immediately displayed the measurements because both containers were using the same <code inline="">/data</code> directory.</p><p>So the observed behaviour is approximately:</p>
Network mode | Web UI | FPing result
-- | -- | --
Docker bridge | Works on :8086 | 100% loss / NaN
Host | Apache conflicts with NAS port 80 | Real measurements appear in RRD

<p>I also performed a full reboot of the Synology and allowed only the original bridge-mode container to start. After the reboot, the FPing graphs were still not receiving measurements.</p><p>This makes me suspect that the problem may be related to ICMP handling/network namespaces on Synology's Docker bridge rather than SmokePing's configuration itself.</p><h2>Question</h2><p>Is there anything specific required by this image for FPing to work correctly through Docker bridge networking on Synology?</p><p>Alternatively, is Host networking expected to be supported, and if so, is there a recommended way to configure the Apache listener to use a different port?</p><p>Lastly, may I suggest making the Apache port configurable somehow? This could be via a file in the config folder or an environment variable during container configuration.</p><p>I would be happy to provide additional logs/configuration if useful and possible on my setup.</p></body></html>
</body>
</html>
